lib/public/Files/Cache/IFileAccess.php

```diff
@@ -79,4 +79,35 @@ public function getByFileIds(array $fileIds): array;
 	 * @since 29.0.0
 	 */
 	public function getByFileIdsInStorage(array $fileIds, int $storageId): array;
+
+	/**
+	 * Retrieves files stored in a specific storage that have a specified ancestor in the file hierarchy.
+	 * Allows filtering by mime types, encryption status, and limits the number of results.
+	 *
+	 * @param int $storageId The ID of the storage to search within.
+	 * @param int $folderId The file ID of the ancestor to base the search on.
+	 * @param int $fileIdCursor The last processed file ID. Only files with a higher ID will be included. Defaults to 0.
+	 * @param int $maxResults The maximum number of results to retrieve. If set to 0, all matching files will be retrieved.
+	 * @param list<int> $mimeTypeIds An array of mime types to filter the results. If empty, no mime type filtering will be applied.
+	 * @param bool $endToEndEncrypted Whether to include EndToEndEncrypted files
+	 * @param bool $serverSideEncrypted Whether to include ServerSideEncrypted files
+	 * @return \Generator A generator yielding matching files as cache entries.
+	 * @throws \OCP\DB\Exception
+	 */
```

Comment:
Why is psalm not red? This is OCP thus requires a `since`

```suggestion
	 *
	 * @since 32.0.0
	 */
```